src/server/middleware/repoproxy/proxy.go

```diff
@@ -173,6 +193,15 @@ func defaultBlobURL(projectName string, name string, digest string) string {
 	return fmt.Sprintf("/v2/%s/library/%s/blobs/%s", projectName, name, digest)
 }
 
+// upstreamRegistryConnectionKey get upstream registry connection key
+func upstreamRegistryConnectionKey(art lib.ArtifactInfo) string {
+	level := os.Getenv(upstreamRegistryLimitLevel)
+	if strings.EqualFold("project", level) {
```

Comment:
Please ensure that the condition only allows upstreamRegistryLimitLevel to be set to either project or artifact.

Reply:
updated